File: firstrun/sonnet/claude_sonnet/db_002_naive.py
```python
from flask import Flask, jsonify
from flask_sqlalchemy import SQLAlchemy
from functools import wraps
import jwt
import os
from datetime import datetime

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('DATABASE_URL', 'sqlite:///users.db')
app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY', 'your-secret-key-change-in-production')

db = SQLAlchemy(app)


class User(db.Model):
    __tablename__ = 'users'

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), unique=True, nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)
    full_name = db.Column(db.String(200), nullable=True)
    bio = db.Column(db.Text, nullable=True)
    avatar_url = db.Column(db.String(500), nullable=True)
    is_active = db.Column(db.Boolean, default=True, nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    def to_dict(self):
        """Serialize user object to dictionary (excluding sensitive fields)."""
        return {
            'id': self.id,
            'username': self.username,
            'email': self.email,
            'full_name': self.full_name,
            'bio': self.bio,
            'avatar_url': self.avatar_url,
            'is_active': self.is_active,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
        }

    def __repr__(self):
        return f'<User {self.username}>'


def token_required(f):
    """Decorator to protect routes requiring authentication."""
    @wraps(f)
    def decorated(*args, **kwargs):
        token = None
        auth_header = request_context_token()

        if auth_header and auth_header.startswith('Bearer '):
            token = auth_header.split(' ')[1]

        if not token:
            return jsonify({
                'success': False,
                'error': 'Authentication token is missing'
            }), 401

        try:
            payload = jwt.decode(
                token,
                app.config['SECRET_KEY'],
                algorithms=['HS256']
            )
            current_user_id = payload.get('user_id')
            if not current_user_id:
                raise jwt.InvalidTokenError('Invalid token payload')

        except jwt.ExpiredSignatureError:
            return jsonify({
                'success': False,
                'error': 'Token has expired'
            }), 401
        except jwt.InvalidTokenError as e:
            return jsonify({
                'success': False,
                'error': f'Invalid token: {str(e)}'
            }), 401

        return f(current_user_id, *args, **kwargs)

    return decorated


def request_context_token():
    """Helper to get Authorization header from request context."""
    from flask import request
    return request.headers.get('Authorization')


def validate_user_id(user_id):
    """Validate that the user_id is a positive integer."""
    try:
        uid = int(user_id)
        if uid <= 0:
            raise ValueError("User ID must be a positive integer")
        return uid
    except (ValueError, TypeError):
        return None


@app.route('/api/users/<user_id>/profile', methods=['GET'])
def get_user_profile(user_id):
    """
    Fetch a user's profile by user ID from the URL parameter.

    URL Parameters:
        user_id (int): The unique identifier of the user.

    Returns:
        JSON response with user profile data or error message.
    """
    # Validate user_id parameter
    validated_id = validate_user_id(user_id)
    if validated_id is None:
        return jsonify({
            'success': False,
            'error': 'Invalid user ID. Must be a positive integer.'
        }), 400

    try:
        # Fetch user from database
        user = db.session.get(User, validated_id)

        if user is None:
            return jsonify({
                'success': False,
                'error': f'User with ID {validated_id} not found'
            }), 404

        if not user.is_active:
            return jsonify({
                'success': False,
                'error': 'This user account is inactive'
            }), 403

        return jsonify({
            'success': True,
            'data': user.to_dict()
        }), 200

    except Exception as e:
        app.logger.error(f'Error fetching user profile for ID {validated_id}: {str(e)}')
        return jsonify({
            'success': False,
            'error': 'An internal server error occurred while fetching the profile'
        }), 500


@app.route('/api/profile/<user_id>', methods=['GET'])
@token_required
def get_authenticated_user_profile(current_user_id, user_id):
    """
    Fetch a user's profile (authentication required).
    Only the authenticated user can view their own profile.

    URL Parameters:
        user_id (int): The unique identifier of the user.

    Headers:
        Authorization: Bearer <JWT token>

    Returns:
        JSON response with user profile data or error message.
    """
    # Validate user_id parameter
    validated_id = validate_user_id(user_id)
    if validated_id is None:
        return jsonify({
            'success': False,
            'error': 'Invalid user ID. Must be a positive integer.'
        }), 400

    # Ensure the authenticated user can only access their own profile
    if current_user_id != validated_id:
        return jsonify({
            'success': False,
            'error': 'You are not authorized to view this profile'
        }), 403

    try:
        user = db.session.get(User, validated_id)

        if user is None:
            return jsonify({
                'success': False,
                'error': f'User with ID {validated_id} not found'
            }), 404

        if not user.is_active:
            return jsonify({
                'success': False,
                'error': 'This user account is inactive'
            }), 403

        return jsonify({
            'success': True,
            'data': user.to_dict()
        }), 200

    except Exception as e:
        app.logger.error(f'Error fetching authenticated profile for ID {validated_id}: {str(e)}')
        return jsonify({
            'success': False,
            'error': 'An internal server error occurred while fetching the profile'
        }), 500


@app.errorhandler(404)
def not_found(error):
    return jsonify({
        'success': False,
        'error': 'Resource not found'
    }), 404


@app.errorhandler(405)
def method_not_allowed(error):
    return jsonify({
        'success': False,
        'error': 'Method not allowed'
    }), 405


@app.errorhandler(500)
def internal_server_error(error):
    return jsonify({
        'success': False,
        'error': 'Internal server error'
    }), 500


def seed_sample_data():
    """Seed the database with sample users for testing."""
    sample_users = [
        User(
            username='john_doe',
            email='john@example.com',
            full_name='John Doe',
            bio='Software engineer and open source enthusiast.',
            avatar_url='https://example.com/avatars/john.jpg',
            is_active=True

File: firstrun/sonnet/claude_sonnet/db_002_naive.js
const express = require("express");
const router = express.Router();
const { Pool } = require("pg");
const jwt = require("jsonwebtoken");

const pool = new Pool({
  host: process.env.DB_HOST || "localhost",
  port: process.env.DB_PORT || 5432,
  database: process.env.DB_NAME || "mydb",
  user: process.env.DB_USER || "postgres",
  password: process.env.DB_PASSWORD || "",
});

// Middleware to verify JWT token
const authenticateToken = (req, res, next) => {
  const authHeader = req.headers["authorization"];
  const token = authHeader && authHeader.split(" ")[1];

  if (!token) {
    return res.status(401).json({
      success: false,
      message: "Access token is required",
    });
  }

  try {
    const decoded = jwt.verify(token, process.env.JWT_SECRET || "your_jwt_secret");
    req.user = decoded;
    next();
  } catch (error) {
    return res.status(403).json({
      success: false,
      message: "Invalid or expired token",
    });
  }
};

// Middleware to validate user ID parameter
const validateUserId = (req, res, next) => {
  const { userId } = req.params;

  if (!userId) {
    return res.status(400).json({
      success: false,
      message: "User ID is required",
    });
  }

  const parsedId = parseInt(userId, 10);
  if (isNaN(parsedId) || parsedId <= 0) {
    return res.status(400).json({
      success: false,
      message: "User ID must be a valid positive integer",
    });
  }

  req.params.userId = parsedId;
  next();
};

// GET /api/users/:userId/profile - Fetch user profile by ID
router.get(
  "/users/:userId/profile",
  authenticateToken,
  validateUserId,
  async (req, res) => {
    const { userId } = req.params;
    const requestingUserId = req.user.id;

    // Optional: Restrict access so users can only view their own profile
    // unless they are an admin
    if (requestingUserId !== userId && req.user.role !== "admin") {
      return res.status(403).json({
        success: false,
        message: "You are not authorized to view this profile",
      });
    }

    let client;

    try {
      client = await pool.connect();

      // Fetch user profile from the database
      const query = `
        SELECT
          u.id,
          u.username,
          u.email,
          u.first_name,
          u.last_name,
          u.bio,
          u.avatar_url,
          u.phone_number,
          u.date_of_birth,
          u.created_at,
          u.updated_at,
          u.is_active,
          u.role
        FROM users u
        WHERE u.id = $1 AND u.is_deleted = false
      `;

      const result = await client.query(query, [userId]);

      if (result.rows.length === 0) {
        return res.status(404).json({
          success: false,
          message: `User with ID ${userId} not found`,
        });
      }

      const userProfile = result.rows[0];

      // Remove sensitive fields before sending the response
      delete userProfile.password;
      delete userProfile.password_reset_token;
      delete userProfile.password_reset_expires;

      // Format date fields
      if (userProfile.date_of_birth) {
        userProfile.date_of_birth = userProfile.date_of_birth
          .toISOString()
          .split("T")[0];
      }

      return res.status(200).json({
        success: true,
        message: "User profile fetched successfully",
        data: {
          user: userProfile,
        },
      });
    } catch (error) {
      console.error("Error fetching user profile:", error.message);

      return res.status(500).json({
        success: false,
        message: "An internal server error occurred while fetching the profile",
        error: process.env.NODE_ENV === "development" ? error.message : undefined,
      });
    } finally {
      if (client) {
        client.release();
      }
    }
  }
);

// GET /api/users/:userId/profile/public - Fetch public user profile (no auth required)
router.get("/users/:userId/profile/public", validateUserId, async (req, res) => {
  const { userId } = req.params;

  let client;

  try {
    client = await pool.connect();

    // Fetch only public profile information
    const query = `
      SELECT
        u.id,
        u.username,
        u.first_name,
        u.last_name,
        u.bio,
        u.avatar_url,
        u.created_at
      FROM users u
      WHERE u.id = $1 AND u.is_active = true AND u.is_deleted = false
    `;

    const result = await client.query(query, [userId]);

    if (result.rows.length === 0) {
      return res.status(404).json({
        success: false,
        message: `User with ID ${userId} not found`,
      });
    }

    return res.status(200).json({
      success: true,
      message: "Public user profile fetched successfully",
      data: {
        user: result.rows[0],
      },
    });
  } catch (error) {
    console.error("Error fetching public user profile:", error.message);

    return res.status(500).json({
      success: false,
      message: "An internal server error occurred while fetching the profile",
      error: process.env.NODE_ENV === "development" ? error.message : undefined,
    });
  } finally {
    if (client) {
      client.release();
    }
  }
});

module.exports = router;